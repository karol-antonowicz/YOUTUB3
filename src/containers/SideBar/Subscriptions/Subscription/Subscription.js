import React from 'react';
import {Icon, Image, Menu} from 'semantic-ui-react';
import './Subscription.scss';

export function Subscription(props) {
    let rightElement = null;    // if there is no new videos, and no broadcasting then nothing is there, otherwise it wud throw an error of undefined I guess
    const {broadcasting, amountNewVideos} = props;
    if (broadcasting) {
        rightElement = <Icon name='bullseye'/>;
    } else if (amountNewVideos) {
        rightElement = <span className='new-videos-count'>{amountNewVideos}</span>;
    }

    return (
        <Menu.Item>
            <div className='subscription'>
                <div>
                    <Image src='http://via.placeholder.com/28x28' avatar/>  
                    <span>{props.label}</span>
                </div>
                {rightElement}
            </div>
        </Menu.Item>

    )
}

