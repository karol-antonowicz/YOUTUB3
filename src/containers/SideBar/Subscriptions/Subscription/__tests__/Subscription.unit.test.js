import React from 'react';
import {shallow} from 'enzyme';
import {Subscription} from '../Subscription';

describe('Subscription', () => {

test('renders Subscription with props=null', ()=>{
    const wrapper = shallow(
        <Subscription/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders Subscription broadcasting, and no new videos', ()=>{
    const wrapper = shallow(
        <Subscription broadcasting label='sample'/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders Subscription broadcasting, and some new videos', ()=>{
    const wrapper = shallow(
        <Subscription broadcasting amountNewVideo={69} label='sample'/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders Subscription not broadcasting, and some new videos', ()=>{
    const wrapper = shallow(
        <Subscription amountNewVideo={69} label='sampledupa'/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders Subscription broadcasting, and no label no new videos', ()=>{
    const wrapper = shallow(
        <Subscription broadcasting/>
    );
    expect(wrapper).toMatchSnapshot();
});

})