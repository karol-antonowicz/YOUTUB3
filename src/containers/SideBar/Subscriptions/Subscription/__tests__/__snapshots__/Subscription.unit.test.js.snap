// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders Subscription broadcasting, and no label no new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
    <Icon
      as="i"
      name="bullseye"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription broadcasting, and no new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        sample
      </span>
    </div>
    <Icon
      as="i"
      name="bullseye"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription broadcasting, and some new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        sample
      </span>
    </div>
    <Icon
      as="i"
      name="bullseye"
    />
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription not broadcasting, and some new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        sampledupa
      </span>
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription renders Subscription with props=null 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;
